savaslabs/savaslabs.github.io
# Issue #396: Test accessibility

@chrisarusso This adds the `gulp-accessibility` plugin and integrates it into our automated tests. I'm targeting most HTML files, with some exceptions:

- No blog posts prior to 2017 (I tested them all locally and am not including them to cut down on the length of time it takes to run the test)
- Only one team page, service page, tag page, etc.
- No pages that are only redirects (e.g. old /news/ pages that redirect to /blog/)

## Notes

- I've ignored a few of the accessibility rules and documented the reasons why in the gulp task comments.
- The test process will stop if any errors are caught but not for warnings. We'll have to keep an eye on these and fix what it makes sense to fix.
- The first commit corrects a couple of issues I found with tag pages that showed up when I looked at the [Durham tag page](http://savaslabs.com/blog/tag/durham/)

## To test

- Pull this code and run `npm install`
- Run `gulp accessibility-test`. You shouldn't see any warnings or errors.
- Open up `_pages/contact.html`. Remove the alt attribute of the image tag. Run `gulp-accessibility` again - you should get an error and the process should stop.
- Check out the following pages that contain updates to correct accessibility issues found by the test:
  - http://localhost:3000/results/open-source/ - removed `<strong>` tags, using CSS to bold text
  - http://localhost:3000/2017/03/06/five-drupal-8-tips-to-empower-content-authors.html - added a label to the honeypot field to tell humans using screenreaders not to fill it out
  - http://localhost:3000 - updated footer social icons to be an unordered list and to have link text hidden by CSS
  - http://localhost:3000/blog/tag/durham/ - updated logic to show or hide the blog post section depending on whether there are any related blog posts. This page has none so this section shouldn't appear.

I also updated heading levels on a bunch of blog posts and other pages, in some cases using CSS to make the headings look the same as before even though the heading element level is different.

Fun fact: `<del></del>` is the new semantic `<s></s>`!
